pithos/pithos.py

```diff
@@ -1051,24 +1054,80 @@ def show_about(self, version):
 
     def on_prefs_response(self, widget, response):
         self.prefs_dlg.hide()
+        new_email = self.prefs_dlg.email_entry.get_text()
+        new_password = self.prefs_dlg.password_entry.get_text()
+        new_proxy = self.prefs_dlg.proxy_entry.get_text()
+        new_control_proxy = self.prefs_dlg.control_proxy_entry.get_text()
+        new_control_proxy_pac = self.prefs_dlg.control_proxy_pac_entry.get_text()
+        new_one_checkbox = self.prefs_dlg.pandora_one_checkbutton.get_active()
 
-        email = self.settings['email']
         if response == Gtk.ResponseType.APPLY:
-            self.on_explicit_content_filter_checkbox()
-            if get_account_password(email) != self.last_pass:
+            if self.last_password != new_password or self.last_email != new_email:
+                #full stop, reset and reconnect on password and/or email change.
```

Comment:
The style for comments is generally one space after the comment and correct capitalization.